src/platform/assets/composables/useMediaAssetActions.ts

```diff
@@ -273,37 +384,73 @@ export function useMediaAssetActions() {
-                  }
-                })
+              // Delete all assets using Promise.allSettled to track individual results
+              const results = await Promise.allSettled(
+                assets.map((asset) =>
+                  deleteAssetApi(asset, getAssetType(asset))
+                )
               )
 
+              // Count successes and failures
+              const succeeded = results.filter(
+                (r) => r.status === 'fulfilled'
+              ).length
+              const failed = results.filter((r) => r.status === 'rejected')
+
+              // Log failed deletions for debugging
+              failed.forEach((result, index) => {
+                console.warn(
+                  `Failed to delete asset ${assets[index].name}:`,
+                  result.reason
+                )
+              })
+
               // Update stores after deletions
-              await assetsStore.updateHistory()
-              if (assets.some((a) => a.tags?.[0] === 'input')) {
+              const hasOutputAssets = assets.some(
+                (a) => getAssetType(a) === 'output'
+              )
+              const hasInputAssets = assets.some(
+                (a) => getAssetType(a) === 'input'
+              )
+
+              if (hasOutputAssets) {
+                await assetsStore.updateHistory()
+              }
+              if (hasInputAssets) {
                 await assetsStore.updateInputs()
               }
 
-              toast.add({
-                severity: 'success',
-                summary: t('g.success'),
-                detail: t('mediaAsset.selection.assetsDeletedSuccessfully', {
-                  count: assets.length
-                }),
-                life: 2000
-              })
+              // Show appropriate feedback based on results
+              if (failed.length === 0) {
+                // All succeeded
+                toast.add({
+                  severity: 'success',
+                  summary: t('g.success'),
+                  detail: t('mediaAsset.selection.assetsDeletedSuccessfully', {
+                    count: succeeded
+                  }),
+                  life: 2000
+                })
+              } else if (succeeded === 0) {
+                // All failed
+                toast.add({
+                  severity: 'error',
+                  summary: t('g.error'),
+                  detail: t('mediaAsset.selection.failedToDeleteAssets'),
+                  life: 3000
+                })
+              } else {
+                // Partial success
+                toast.add({
+                  severity: 'warn',
+                  summary: t('g.warning'),
+                  detail: t('mediaAsset.selection.partialDeleteSuccess', {
+                    succeeded,
+                    failed: failed.length
+                  }),
+                  life: 3000
+                })
+              }
             } catch (error) {
```

Comment:
half-nit, half-actual-concern:
When there are so many things happening in the try block, it's easy for the operations to have partially completed which can leave the store in an invalid state.
Could you use multiple smaller try blocks, maybe tracking which pieces specifically failed?